helix-term/src/commands.rs

```diff
@@ -1032,15 +1033,24 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
         .map(|r| text.slice(r.from()..r.to()).to_string())
         .collect();
     let primary = selections.primary();
-    if selections.len() == 1 && primary.to() - primary.from() == 1 {
-        let current_word = movement::move_next_long_word_start(
+    // Checks whether there is only one selection with a width of 1
+    if selections.len() == 1 && primary.len() == 1 {
+        let count = cx.count();
+        // In this case it selects the WORD under the cursor
+        let current_word = textobject::textobject_word(
             text.slice(..),
-            movement::move_prev_long_word_start(text.slice(..), primary, 1),
-            1,
+            primary,
+            textobject::TextObject::Inside,
+            count,
+            true,
         );
+        // Trims some surrounding chars so that the actual file is opened.
+        let surrounding_chars: &[_] = &['\'', '"', '(', ')'];
         paths.clear();
         paths.push(
             text.slice(current_word.from()..current_word.to())
+                .to_string()
```

Comment:
We can use `current_word.fragment(text)` and then `.trim_matches` on that - it will have a chance of not copying the range's contents until we get to the `.to_string()` at the bottom (calling `.to_string()` will always copy the contents).

(For this to work, we'll need to add a line up on L1038 saying `let text = text.slice(..)` and then we can re-use the ropeslice in the call to `textobject_word`.)

Reply:
Thank you very much!